torch/_subclasses/fake_tensor.py

```diff
@@ -1420,7 +1445,10 @@ def run_fallback_kernel(fake_mode, func, args, kwargs, orig_not_implemented_exce
 
         def to_real_tensor(e):
             if isinstance(e, FakeTensor):
-                out = torch.zeros_like(e, device=e.fake_device)
+                if e.is_nested:
+                    out = torch.empty_like(e, device=e.fake_device)
```

Comment:
Do you know if fake prop fallback even works with nested tensor?

Reply:
It works - added a test to hit that path